intranet/static/js/common.js

```diff
@@ -31,25 +30,159 @@ $(function() {
+                        udlr.input = "";
+                        e.preventDefault();
+                        return false
+                    } else if (udlr.input === udlr.pattern.substr(0, udlr.input.length)) e.preventDefault();
+                    else udlr.input = "";
+                }, this);
+                this.iphone.load(link)
+            },
+            code: function(link) {
+                window.location = link
+            },
+            iphone: {
+                start_x: 0,
+                start_y: 0,
+                stop_x: 0,
+                stop_y: 0,
+                tap: false,
+                capture: false,
+                orig_keys: "",
+                keys: ["UP", "UP", "DOWN", "DOWN", "LEFT", "RIGHT", "LEFT", "RIGHT", "TAP", "TAP"],
+                code: function(link) {
+                    udlr.code(link)
+                },
+                load: function(link) {
+                    this.orig_keys = this.keys;
+                    udlr.addEvent(document, "touchmove", function(e) {
+                        if (e.touches.length === 1 && udlr.iphone.capture === true) {
+                            var touch = e.touches[0];
+                            udlr.iphone.stop_x = touch.pageX;
+                            udlr.iphone.stop_y = touch.pageY;
+                            udlr.iphone.tap = false;
+                            udlr.iphone.capture = false;
+                            udlr.iphone.check_direction()
+                        }
+                    });
+                    udlr.addEvent(document, "touchend", function(evt) {
+                        if (udlr.iphone.tap === true) udlr.iphone.check_direction(link)
+                    }, false);
+                    udlr.addEvent(document, "touchstart", function(evt) {
+                        udlr.iphone.start_x = evt.changedTouches[0].pageX;
+                        udlr.iphone.start_y = evt.changedTouches[0].pageY;
+                        udlr.iphone.tap = true;
+                        udlr.iphone.capture = true
+                    })
+                },
+                check_direction: function(link) {
+                    x_magnitude = Math.abs(this.start_x - this.stop_x);
+                    y_magnitude = Math.abs(this.start_y - this.stop_y);
+                    x = ((this.start_x - this.stop_x) < 0) ? "RIGHT" : "LEFT";
+                    y = ((this.start_y - this.stop_y) < 0) ? "DOWN" : "UP";
+                    result = (x_magnitude > y_magnitude) ? x : y;
+                    result = (this.tap == true) ? "TAP" : result;
+                    if (result == this.keys[0]) this.keys = this.keys.slice(1, this.keys.length);
+                    if (this.keys.length == 0) {
+                        this.keys = this.orig_keys;
+                        this.code(link)
+                    }
+                }
+            }
+        };
+        typeof callback === "string" && udlr.load(callback);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/tjcsl/ion/file/2743307438/issues/source?bid=3176160&fileBranchId=3355598#l135)